test/hotspot/gtest/unittest.hpp

```diff
@@ -135,7 +135,7 @@
   TEST(category, CONCAT(name, _vm_assert)) {                        \
     ASSERT_EXIT(child_ ## category ## _ ## name ## _(),             \
                 ::testing::ExitedWithCode(1),                       \
-                "^assert failed: " msg);                            \
+                "^assert failed: .*" msg);                          \
```

Comment:
So if I take out '^' here, I've also fixed https://bugs.openjdk.java.net/browse/JDK-8257229 ?  I had a version without ^.

Reply:
I think it's better to fix JDK-8257229 separately, so it's easier to track the issues.

Adding `.*` here affects all tests, and it might make the match more liberal than necessary. E.g., if we have `msg = "end of the world"`, the regexp would inadvertently match

```
assert failed: some unrelated assert
Whew, we avoided the end of the world
```

It would be better to add the .* in a specific test like 

```
TEST_VM_ASSERT_MSG(vmErrorTest, assert1, ".*assert.str == NULL. failed: expected null") {
```

Reply:
A better example for not adding the `.*` :  `TEST_VM_ASSERT_MSG(...., "a == true")` would incorrectly match `assert failed: !a == true`

Reply:
I reviewed your change.  There are 19 tests that only want to test the message, which seems legitimate.

Reply:
I fixed he other tests with a combination of adding the assert or adding a ".*" in front of the expected message.